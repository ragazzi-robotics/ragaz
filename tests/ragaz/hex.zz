def main():

    var a = 0xff
    print(a)

    # range: 0 to 1
    print(hex(False))
    print(hex(True))

    # range: 0 to 256
    print(hex(0 as byte))
    print(hex(1 as byte))

    # range: -128 to 127
    #TODO: Make the code bellow produce: 0x7f
    #print(hex(-127 as i8))
    print(hex(128 as i8))

    # range: 0 to 255
    print(hex(0 as u8))
    print(hex(255 as u8))

    # -32,768 to 32,767
    print(hex(-32768 as i16))
    print(hex(32767 as i16))

    # 0 to 65,535
    print(hex(0 as u16))
    print(hex(65535 as u16))

    # -2,147,483,648 to 2,147,483,647
    print(hex(-2147483648 as i32))
    print(hex(2147483647 as i32))

    # 0 to 4,294,967,295
    print(hex(0 as u32))
    print(hex(4294967295 as u32))

    # -2,147,483,648 to 2,147,483,647
    print(hex(-2147483648 as i64))
    print(hex(2147483647 as i64))

    # 0 to 4,294,967,295
    print(hex(0 as u64))
    print(hex(4294967295 as u64))

    # -(2^63) to (2^63)-1
    #TODO: Make the code bellow produce: 0x8000000000000000
    #print(hex((-(2**63)) as i128))
    print(hex(((2^63)-1) as i128))

    # 0 to 18,446,744,073,709,551,615
    print(hex(0 as u128))
    #TODO: Make the code bellow produce: 0xffffffffffffffff
    #print(hex(18446744073709551615 as u128))

    # -2,147,483,648 to 2,147,483,647
    print(hex(-2147483648 as int))
    print(hex(2147483647 as int))

    # 0 to 4,294,967,295
    print(hex(0 as uint))
    print(hex(4294967295 as uint))
