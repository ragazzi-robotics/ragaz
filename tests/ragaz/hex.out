255
0x0
0x1
0x0
0x1
0x80
0x0
0xff
0x8000
0x7fff
0x0
0xffff
0x80000000
0x7fffffff
0x0
0xffffffff
0x80000000
0x7fffffff
0x0
0xffffffff
0x3c
0x0
0x80000000
0x7fffffff
0x0
0xffffffff
